app/factory/Layer.js

```diff
@@ -67,6 +67,12 @@ Ext.define('CpsiMapview.factory.Layer', {
         case 'switchlayer':
             mapLayer = LayerFactory.createSwitchLayer(layerConf);
             break;
+        case 'vt':
+            mapLayer = LayerFactory.createVectorTilesLayer(layerConf);
+            break;
+        case 'vtwms':
+            mapLayer = LayerFactory.createVectorTilesWmsLayer(layerConf);
+            break;
```

Comment:
Will any of the new source somehow be able to throw the custum events around loading / progress (e.g. `vectorloadstart` and friends). If I read the diff correctly, right now they won't, but I really do not know, if we can change this so they can (with a loader function or sth.)

Reply:
Would already work since appropriate events are fired in the source. The `MapLoadingStatusBar` of BasiGX does currently not handle them. I opened a PR in the BasiGX repo: https://github.com/terrestris/BasiGX/pull/504
Once this is merged the `MapLoadingStatusBar` will show the correct progress.